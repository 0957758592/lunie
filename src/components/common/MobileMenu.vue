<template>
  <v-footer :padless="true" :fixed="true" dark>
    <v-card flat tile width="100%" class="text-center">
      <v-card-text>
        <v-btn
          v-for="link in links"
          :key="link.icon"
          icon
          exact
          small
          :to="link.route"
        >
          <v-icon size="24px">{{ link.icon }}</v-icon>
          <!-- <span>{{ link.title }}</span> -->
        </v-btn>
      </v-card-text>
    </v-card>
  </v-footer>
</template>

<script>
export default {
  name: `mobile-menu`,
  props: {
    links: {
      type: Array,
      required: true
    }
  }
}
</script>

<style scoped>
.v-footer, .v-card {
    background: var(--app-nav);
    color: var(--bright);
  text-align: center;
}
.v-card__text {
  padding: 0.5rem;
}
.v-btn:not(:last-child) {
  margin-right: 1.5rem;
}
</style>
