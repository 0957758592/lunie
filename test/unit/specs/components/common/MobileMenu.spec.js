import { shallowMount } from "@vue/test-utils"
import MobileMenu from "common/MobileMenu"

describe(`MobileMenu`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(MobileMenu, {
      props: {
        links: [
          {
            icon: "account_balance_wallet",
            route: "/portfolio",
            title: "Portfolio"
          },
          {
            icon: "add_box",
            route: "/validators",
            title: "Validators"
          }
        ]
      },
      stubs: [
        `router-link`,
        `v-footer`,
        `v-divider`,
        `v-card`,
        `v-card-text`,
        `v-navigation-drawer`,
        `v-list`,
        `v-list-item`,
        `v-list-item-title`,
        `v-list-item-subtitle`,
        `v-list-item-content`,
        `v-list-item-avatar`,
        `v-icon`,
        `v-list-item-icon`
      ]
    })
  })

  it("should show a mobile menu", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
