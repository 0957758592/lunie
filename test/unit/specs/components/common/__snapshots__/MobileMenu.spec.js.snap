// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu should show a mobile menu 1`] = `
<v-footer
  dark=""
  fixed="true"
  padless="true"
>
  <v-card
    class="text-center"
    flat=""
    tile=""
    width="100%"
  >
    <v-card-text />
  </v-card>
</v-footer>
`;
